sky/engine/core/window/window.cc

```diff
@@ -14,6 +14,12 @@
 namespace blink {
 namespace {
 
+// These strings must match the corresponding strings in hooks.dart
+const std::map<sky::AppLifecycleState, const char *> kLifecycleStateMap = {
```

Comment:
I think I'd just pass the numeric value of the enum on the c++ side, and have a lookup table on the dart side keyed on int.